Composed/Core/DataSources/SingleElementDataSource.swift

```diff
@@ -0,0 +1,101 @@
+/**
+ A `DataSource` that contains a single element. This DataSource will always
+ return `1` for `numberOfSections`.
+ */
+open class SingleElementDataSource<Element>: DataSource {
+
+    /// A delegate that will be notified of updates to the data source
+    public weak var updateDelegate: DataSourceUpdateDelegate?
+    
+    /// The element being stored by the data source
+    public private(set) var element: Element
+    
+    /// `true` if `element` is an `Optional.none`
+    public var elementIsNil: Bool {
+        switch element as Any {
+        case Optional<Any>.none: return true
+        default: return false
+        }
+    }
+
+    /// The number of elements the data source provides. Hardcoded to `1`
+    public var numberOfSections: Int { return 1 }
+
+    /// The index paths this data source provides. Each `IndexPath` will
+    /// have a section of 0
+    public var indexPaths: [IndexPath] {
```

Comment:
Not sure the point of this and to be honest, this is a single element, it only ever has 1 indexPath which is always 0, 0
The API consumer doesn't have to use `element(at indexPath:)` in fact this dataSource doesn't even have that function. So they'd always just grab the `element` directly. The indexPath has very little value, but if they did want to have access to it, I'd imagine again its easier for the consumer to just add an extension:
```swift
var indexPath: IndexPath { return IndexPath(item: 0, section: 0) }
```